src/utils/pricing_utils.ts

```diff
@@ -97,9 +99,20 @@ export class PlanLimitsHandler {
 			if (this.repoId) return await this.run();
 			// This is "Connect Repo"
 			const isNewPrivateRepo = errorCode === ErrorCodes.PRIVATE_REPO_COUNT_LIMIT_REACHED;
-			showFreeTierLimitReached(this.repoPath, isNewPrivateRepo);
+			const canAvailTrial = await getCanAwaitTrial(this.accessToken);
+			showFreeTierLimitReached(this.repoPath, isNewPrivateRepo, canAvailTrial);
 			return true;
 		}
 		return false;
 	}	
 }
+
+export const getCanAwaitTrial = async(accessToken: string) : Promise<IUserSubscriptionInfo> => {
+	// Get CanAvailTrial from server
+	const userSubscriptionInfo = <IUserSubscriptionInfo>{};
+	const json = <any> await getUserSubcription(accessToken);
+	if (!json.error) {
+		userSubscriptionInfo.canAvailTrial = json?.response?.subscription?.can_avail_trial;
+	}
+	return userSubscriptionInfo;
+};
```

Comment:
What if there is an error? Are we logging somewhere or what are we returning in the case of error

Reply:
Good observation, I added logging in case there is an error: 

![image](https://github.com/codesyncapp/codesync-vs/assets/15974895/75bab832-633f-4b37-8c63-4892a0f40383)